parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java

```diff
@@ -558,30 +556,38 @@ public SELF withPageWriteChecksumEnabled(boolean enablePageWriteChecksum) {
     }
 
     /**
-     * Specified the column and the NDV map for the bloom filter.
+     * Sets the NDV (number of distinct values) for the specified column.
      *
-     * @param columnNDVMap the column to NDV map
+     * @param columnPath the path of the column (dot-string)
+     * @param ndv        the NDV of the column
      *
      * @return this builder for method chaining.
      */
-    public SELF withBloomFilterColumnToNDVMap(Map<String, Long> columnNDVMap) {
-      if (columnNDVMap != null) {
-        encodingPropsBuilder.withBloomFilterColumnToNDVMap(columnNDVMap);
-      }
-
+    public SELF withBloomFilterNDV(String columnPath, long ndv) {
+      encodingPropsBuilder.withBloomFilterNDV(columnPath, ndv);
       return self();
     }
 
     /**
-     * Specified the column names for the bloom filter.
+     * Sets the bloom filter enabled/disabled
      *
-     * @return this builder for method chaining.
+     * @param enabled whether to write bloom filters
+     * @return this builder for method chaining
      */
-    public SELF withBloomFilterColumnNames(String...columns) {
-      if (columns != null) {
-        encodingPropsBuilder.withBloomFilterColumnNames(Arrays.asList(columns));
-      }
+    public SELF withBloomFilterEnabled(boolean enabled) {
+      encodingPropsBuilder.withBloomFilterEnabled(enabled);
+      return self();
+    }
 
+    /**
+     * Sets the bloom filter enabled/disabled for the specified column
```

Comment:
Wouldn't it be better to say that the result of this method will be combined with `withBloomFilterEnabled(boolean enabled)`?

Reply:
I wouldn't say combined. If an enabled/disabled state is set for a specific column then that state would be used independently from the "universal" state set. Let me try to make it more clear in the comments.

Reply:
I agree, it's better now!